### PR TITLE
Add org.lvnauth.LVNAuth

### PR DESCRIPTION
**Pull request by jrezai**
### Please confirm your submission meets all the criteria

LVNAuth is a desktop application that lets users create and play visual novels.

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: org.lvnauth.LVNAuth.json [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                "install -D flatpak/lvnauth_64x64.png /app/share/icons/hicolor/64x64/apps/${FLATPAK_ID}.png",
+                "install -D flatpak/lvnauth_128x128.png /app/share/icons/hicolor/128x128/apps/${FLATPAK_ID}.png",
+                "install -D flatpak/lvnauth_256x256.png /app/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png"
+	    ],
+	    "buildsystem": "simple",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/jrezai/lvnauth.git",
+                    "branch": "main"
```
Comment (hfiguiere) [inline org.lvnauth.LVNAuth.json:35]: never use git branches.

File: org.lvnauth.LVNAuth.json [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                }
+            ]
+        },
+	"python3-tkinter.json",
+        {
+            "name": "cpython",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://www.python.org/ftp/python/3.11.2/Python-3.11.2.tar.xz",
```
Comment (hfiguiere) [inline org.lvnauth.LVNAuth.json:45]: python is in the runtime. why do you need this?

File: org.lvnauth.LVNAuth.json [hunk 3]
```diff
@@ -0,0 +1,135 @@
+{
+    "app-id": "org.lvnauth.LVNAuth",
+    "runtime": "org.freedesktop.Sdk",
+    "runtime-version": "23.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "startup.sh",
+    "finish-args": [
+	    "--socket=x11",
+	    "--socket=fallback-x11",
```
Comment (hfiguiere) [inline org.lvnauth.LVNAuth.json:9]: thse two are exclusive, the latter being used in combination of wayland. So it doesn't support wayland remove fallback.

File: org.lvnauth.LVNAuth.json (hunk 1 above)
Comment (jrezai) [reply org.lvnauth.LVNAuth.json:35]: I'll work on fixing this, thanks.

File: org.lvnauth.LVNAuth.json (hunk 2 above)
Comment (jrezai) [reply org.lvnauth.LVNAuth.json:45]: Ah ok, I didn't know Python was in the runtime. I'll remove that source and test it. Thanks.

**Comment by hfiguiere** [comment]
bot, build org.lvnauth.LVNAuth

File: org.lvnauth.LVNAuth.json [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            "build-commands": [
+		"install -D flatpak/startup.sh ${FLATPAK_DEST}/bin/startup.sh",
+                "chmod +w ${FLATPAK_DEST}/bin/startup.sh",
+                "cp -r src/* ${FLATPAK_DEST}/bin",
+                "install -D flatpak/org.lvnauth.LVNAuth.desktop /app/share/applications/${FLATPAK_ID}.desktop",
+                "install -D flatpak/lvnauth_16x16.png /app/share/icons/hicolor/16x16/apps/${FLATPAK_ID}.png",
+                "install -D flatpak/lvnauth_32x32.png /app/share/icons/hicolor/32x32/apps/${FLATPAK_ID}.png",
+                "install -D flatpak/lvnauth_64x64.png /app/share/icons/hicolor/64x64/apps/${FLATPAK_ID}.png",
+                "install -D flatpak/lvnauth_128x128.png /app/share/icons/hicolor/128x128/apps/${FLATPAK_ID}.png",
+                "install -D flatpak/lvnauth_256x256.png /app/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png"
```
Comment (hfiguiere) [inline org.lvnauth.LVNAuth.json:27]: I don't see any appstream file installed here. This is why the build did fail.

File: org.lvnauth.LVNAuth.json [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    "command": "startup.sh",
+    "finish-args": [
+	    "--socket=x11",
+	    "--share=network",
+	    "--share=ipc",
+	    "--socket=pulseaudio",
+	    "--filesystem=home",
+	    "--device=dri"
+        ],
+    "modules": [
```
Comment (bbhtt) [inline org.lvnauth.LVNAuth.json:15]:
```suggestion
     ],
    "modules": [
```

File: org.lvnauth.LVNAuth.json [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        {
+            "name": "lvnauth",
+            "build-commands": [
+		"install -D flatpak/startup.sh ${FLATPAK_DEST}/bin/startup.sh",
+                "chmod +w ${FLATPAK_DEST}/bin/startup.sh",
```
Comment (bbhtt) [inline org.lvnauth.LVNAuth.json:20]:
```suggestion
            "build-commands": [
		            "install -D flatpak/startup.sh ${FLATPAK_DEST}/bin/startup.sh",
                "chmod +w ${FLATPAK_DEST}/bin/startup.sh",
```

File: org.lvnauth.LVNAuth.json [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+                "cp -r src/* ${FLATPAK_DEST}/bin",
+                "install -D flatpak/org.lvnauth.LVNAuth.desktop /app/share/applications/${FLATPAK_ID}.desktop",
+                "install -D flatpak/lvnauth_16x16.png /app/share/icons/hicolor/16x16/apps/${FLATPAK_ID}.png",
+                "install -D flatpak/lvnauth_32x32.png /app/share/icons/hicolor/32x32/apps/${FLATPAK_ID}.png",
+                "install -D flatpak/lvnauth_64x64.png /app/share/icons/hicolor/64x64/apps/${FLATPAK_ID}.png",
+                "install -D flatpak/lvnauth_128x128.png /app/share/icons/hicolor/128x128/apps/${FLATPAK_ID}.png",
```
Comment (bbhtt) [inline org.lvnauth.LVNAuth.json:26]: Unless you need the smaller sizes, just install 256px and drop the rest

File: org.lvnauth.LVNAuth.json [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/jrezai/lvnauth.git",
+                    "commit": "76d832ff64c31fc308b5d296087421a907664387"
+                }
+            ]
+        },
+	"python3-tkinter.json",
+        {
```
Comment (bbhtt) [inline org.lvnauth.LVNAuth.json:39]:
```suggestion
        },
	      "python3-tkinter.json",
        {
```

File: org.lvnauth.LVNAuth.json [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+            "name": "flit-core",
```
Comment (bbhtt) [inline org.lvnauth.LVNAuth.json:40]: If you are generating this and below with flatpak-pip-generator it should be in a seperate manifest

File: org.lvnauth.LVNAuth.json (hunk 7 above)
Comment (hfiguiere) [reply org.lvnauth.LVNAuth.json:26]: it's ok to install them when there are already there.

File: org.lvnauth.LVNAuth.json (hunk 4 above)
Comment (jrezai) [reply org.lvnauth.LVNAuth.json:27]: When I add this to the json file:

`"install -D flatpak/org.lvnauth.LVNAuth.metainfo.xml /app/share/metainfo/${FLATPAK_ID}.metainfo.xml",`

I get an error when I attempt to test it using:
`flatpak run org.flatpak.Builder --install-deps-from=flathub --user --install --force-clean build-dir org.lvnauth.LVNAuth.json`

```
Errors were raised during this compose run:
general
  E: filters-but-no-output

org.lvnauth.LVNAuth
  E: metainfo-parsing-error
Refer to the generated issue report data for details on the individual problems.
Error: ERROR: appstreamcli compose failed: Child process exited with code 1

```

File: org.lvnauth.LVNAuth.json (hunk 9 above)
Comment (jrezai) [reply org.lvnauth.LVNAuth.json:40]: Which part(s) need to be in a separate manifest?

Comment (bbhtt) [reply org.lvnauth.LVNAuth.json:40]: The modules starting from this and below

File: org.lvnauth.LVNAuth.json (hunk 4 above)
Comment (hfiguiere) [reply org.lvnauth.LVNAuth.json:27]: does it validate?

still not installed. so the build won't pass.

File: org.lvnauth.LVNAuth.metainfo.xml
Comment (hfiguiere) [inline org.lvnauth.LVNAuth.metainfo.xml:1]: you should use and install the file upstream and not have one here.

File: org.lvnauth.LVNAuth.json (hunk 4 above)
Comment (jrezai) [reply org.lvnauth.LVNAuth.json:27]: It validates now, I had a missing `</release>` section in the metainfo file and some invalid characters. I fixed them yesterday.
I used flatpak-builder-lint to check it.

File: org.lvnauth.LVNAuth.json (hunk 9 above)
Comment (bbhtt) [reply org.lvnauth.LVNAuth.json:40]: You didn't remove them from the main manifest

Comment (jrezai) [reply org.lvnauth.LVNAuth.json]: I forgot to push the change it seems, I just did that, thanks.

**Comment by hfiguiere** [comment]
bot, build org.lvnauth.LVNAuth

File: org.lvnauth.LVNAuth.json [hunk 10]
```diff
@@ -0,0 +1,41 @@
+{
+    "app-id": "org.lvnauth.LVNAuth",
+    "runtime": "org.freedesktop.Sdk",
```
Comment (bbhtt) [inline org.lvnauth.LVNAuth.json:3]: Why are you using the SDK as runtime?

**Comment by bbhtt** [comment]
Creating a new project, choosing location and hitting save, doesn't work:

```sh
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/lib/python3.11/site-packages/tkinter/__init__.py", line 1892, in __call__
    return self.func(*args)
           ^^^^^^^^^^^^^^^^
  File "/app/bin/new_project_window.py", line 150, in on_ok_button_clicked
    messagebox.showerror(parent=self.project_folder_window,
  File "/app/lib/python3.11/site-packages/tkinter/messagebox.py", line 98, in showerror
    return _show(title, message, ERROR, OK, **options)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/tkinter/messagebox.py", line 76, in _show
    res = Message(**options).show()
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/tkinter/commondialog.py", line 46, in show
    s = w.tk.call(self.command, *w._options(self.options))
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_tkinter.TclError: bad option "-messagebox": must be -default, -detail, -icon, -message, -parent, -title, or -type
test@fedora:~$ 
```

**Comment by bbhtt** [comment]
OK, the issue was I had a file named `test` in that folder, so it couldn't create the `test` folder. Something to handle for later.

**Comment by jrezai** [comment]
> OK, the issue was I had a file named `test` in that folder, so it couldn't create the `test` folder. Something to handle for later.

Fixed the message box not showing an error pop-up upon creating new project when it was supposed to (5599d3c)

File: org.lvnauth.LVNAuth.json [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+	    "--socket=pulseaudio",
+	    "--filesystem=home",
+	    "--device=dri"
+     ],
+    "modules": [
+        {
+            "name": "lvnauth",
+            "build-commands": [
+		            "install -D flatpak/startup.sh ${FLATPAK_DEST}/bin/startup.sh",
+                            "chmod +w ${FLATPAK_DEST}/bin/startup.sh",
```
Comment (bbhtt) [inline org.lvnauth.LVNAuth.json:20]: image

Something is wrong with your editor the indentation is getting messed up

**Comment by bbhtt** [comment]
bot, build org.lvnauth.LVNAuth

**Pull request closed without merging**